activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessRuntimeImpl.java

```diff
@@ -297,6 +301,7 @@ public void signal(SignalPayload signalPayload) {
         //@TODO: define security policies for signalling
         runtimeService.signalEventReceived(signalPayload.getName(),
                                            signalPayload.getVariables());
+        eventPublisher.publishEvent(signalPayload);
```

Comment:
@daisuke-yoshimoto The signal() business method is the transactional boundary, which we basically want fully succeed or fully fail not partially succeed. So, it makes sense to define `@Transactional` annotation at the ProcessRuntime service layer and remove the `@Transactional` annotation from sendSignal() method in ProcessInstanceControllerImpl class.

Reply:
@igdianov 
Thanks for your review. I fixed. Please review again.
https://github.com/Activiti/Activiti/pull/2229/commits/9437d7d7474bf71793ff437996025097894b3104
https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/171/commits/57da80954874de6926e8eed25031af8c42d1b1f1